controllers/sandbox_controller.go

```diff
@@ -76,22 +79,96 @@ func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
+	oldStatus := sandbox.Status.DeepCopy()
+
 	// Create a hash from the sandbox.Name and use it as label value
 	nameHash := NameHash(sandbox.Name)
-	_, err := r.reconcilePod(ctx, sandbox, nameHash)
-	if err != nil {
-		return ctrl.Result{}, err
+
+	var allErrors error
+
+	// Reconcile Pod
+	pod, err := r.reconcilePod(ctx, sandbox, nameHash)
+	allErrors = errors.Join(err)
+
+	// Reconcile Service
+	svc, err := r.reconcileService(ctx, sandbox, nameHash)
+	allErrors = errors.Join(allErrors, err)
+
+	// compute and set overall Ready condition
+	readyCondition := r.computeReadyCondition(sandbox.Generation, allErrors, svc, pod)
+	meta.SetStatusCondition(&sandbox.Status.Conditions, readyCondition)
+
+	// Update status
+	err = r.updateStatus(ctx, oldStatus, sandbox)
+	allErrors = errors.Join(allErrors, err)
+
+	// return errors seen
+	return ctrl.Result{}, allErrors
+}
+
+func (r *SandboxReconciler) computeReadyCondition(generation int64, err error, svc *corev1.Service, pod *corev1.Pod) metav1.Condition {
+	readyCondition := metav1.Condition{
+		Type:               string(sandboxv1alpha1.SandboxConditionReady),
+		ObservedGeneration: generation,
+		Message:            "",
+		Status:             metav1.ConditionFalse,
+		Reason:             "DependenciesNotReady",
 	}
 
-	_, err = r.reconcileService(ctx, sandbox, nameHash)
 	if err != nil {
-		return ctrl.Result{}, err
+		readyCondition.Reason = "ReconcilerError"
+		readyCondition.Message = "Error seen: " + err.Error()
+		return readyCondition
 	}
 
-	// Update the sandbox status
-	// TODO: Update the sandbox status
+	readyCount := 0
+
+	message := "Pod or Service not ready"
+	if pod != nil {
+		message = "Pod exists with phase: " + string(pod.Status.Phase)
+		// Check if pod Ready condition is true
+		if pod.Status.Phase == corev1.PodRunning {
+			message = "Pod is Running but not Ready"
+			for _, condition := range pod.Status.Conditions {
+				if condition.Type == corev1.PodReady {
+					if condition.Status == corev1.ConditionTrue {
+						message = "Pod is Ready"
+						readyCount++
+					}
+					break
+				}
+			}
+		}
+	}
+	if svc != nil {
+		message += "; Service Exists"
+		readyCount++
+	}
+
+	readyCondition.Message = message
+	if readyCount == 2 {
```

Comment:
`readyCount` is less reliable, which might produce false negative when we introduce more conditions later. Would you change it to `bool sandboxReady` (default false) that only gets set to `true` when all the resources are ready?